File: WASHHAND.cpp
#include<bits/stdc++.h>
#define ll long long
using namespace std;

int main() {
	ios::sync_with_stdio(0);
	cin.tie(0); cout.tie(0);

	int t;
	cin >> t;
	while(t--) {
		ll n, d;
		string po;
		cin >> n;
		cin >> po;
		cin >> d;
		vector<char> pop;
		for(int i = 0; i < n; i++) {
			pop.push_back(po[i]);
		} 
		vector<int> dl(d);
		for(auto& i1: dl) cin >> i1;
		vector<char> status(n, 'N');
		//memset(&status[0], 'N', sizeof(status[0]) * status.size());  
		for(int i = 0; i < d; i++) {
			status[i - 1] = 'I';
			if(pop[0] == '1' && status[1] == 'N') pop[1] = '1';
			vector<int> temp;
			for(int j = 0 ; j < n; j++) {
				if(pop[j] == '1') temp.push_back(j);
			}
			for(int j = 0; j < temp.size(); j++) {
				if(pop[temp[j]] == '1') {
					if(status[temp[j]] == 'N') pop[temp[j] - 1] = '1';
					if(status[temp[j] + 1] == 'N') pop[temp[j] + 1] = '1';
				}	
			}
			if(pop[-1] == '1' && status[-1] == 'N') pop[-2] = '1'; 
		}
		cout << count(pop.begin(), pop.end(), '1') << "\n";
	} 
	return 0;
}

File: WASHHAND.py
"""

    Author - AYUSHMAN CHAHAR #

"""
from __future__ import division, print_function
import bisect
import math
import heapq
import itertools
import sys
# import numpy
from itertools import combinations
from itertools import permutations
from collections import deque
from atexit import register
from collections import Counter
from functools import reduce

sys.setrecursionlimit(10000000)

if sys.version_info[0] < 3:
    from io import BytesIO as stream
else:
    from io import StringIO as stream

if sys.version_info[0] < 3:
    class dict(dict):
        """dict() -> new empty dictionary"""

        def items(self):
            """D.items() -> a set-like object providing a view on D's items"""
            return dict.iteritems(self)

        def keys(self):
            """D.keys() -> a set-like object providing a view on D's keys"""
            return dict.iterkeys(self)

        def values(self):
            """D.values() -> an object providing a view on D's values"""
            return dict.itervalues(self)


    input = raw_input
    range = xrange

    filter = itertools.ifilter
    map = itertools.imap
    zip = itertools.izip


def sync_with_stdio(sync=True):
    """Set whether the standard Python streams are allowed to buffer their I/O.
 
    Args:
        sync (bool, optional): The new synchronization setting.
 
    """
    global input, flush

    if sync:
        flush = sys.stdout.flush
    else:
        sys.stdin = stream(sys.stdin.read())
        input = lambda: sys.stdin.readline().rstrip('\r\n')

        sys.stdout = stream()
        register(lambda: sys.__stdout__.write(sys.stdout.getvalue()))


def main():
    for _ in range(int(input())):
        n = int(input())
        pop = list(input())
        d = int(input())
        dl = list(map(int, input().split()))
        status = ["NI"] * n

        for i in dl:
            status[i - 1] = 'I'
            if pop[0] == '1' and status[1] == 'NI':
                pop[1] = '1'
            temp = []
            for i in range(1, n - 1):
                if pop[i] == '1':
                    temp.append(i)
            for i in temp:
                if pop[i] == '1':
                    if status[i] == 'NI':
                        pop[i - 1] = '1'
                    if status[i + 1] == 'NI':
                        pop[i + 1] = '1'
            if pop[-1] == '1' and status[-1] == 'NI':
                pop[-2] = '1'
        print(pop.count('1'))


if __name__ == '__main__':
    sync_with_stdio(False)
    main()
